src/core/GameStateManager.ts

```diff
@@ -1,41 +1,82 @@
-import { GameState } from "constants/enums";
+import { Color, GameState } from "constants/enums";
 import Board from "./Board";
 import Move from "./Move";
+import MoveGenerator from "./MoveGenerator";
 import King from "./pieces/King";
+const _ = require('lodash');
 
 class GameStateManager {
-    board: Board;
     state: GameState;
+    moveGenerator: MoveGenerator;
 
-    constructor(board: Board) {
-        this.board = board;
+    constructor(moveGenerator: MoveGenerator) {
+        this.moveGenerator = moveGenerator;
         this.state = GameState.Default;
     }
 
-    getBoardState(king: King, enemyMoves: Move[]): GameState {
-        this.updateState(king, enemyMoves);
+    getBoardState(board: Board, king: King): GameState {
+        this.updateState(board, king);
         return this.state;
     }
 
-    private updateState(king: King, enemyMoves: Move[]): void {
+    private updateState(board: Board, king: King): void {
+        /* Color of the enemy of the considered king */
+        const enemyColor: Color = king.color === Color.White? Color.Black : Color.White;
+
+        const enemyMoves: Move[] = this.moveGenerator.generateForColor(enemyColor);
+        const allowedMoves: Move[] = this.moveGenerator.generateForColor(king.color);
```

Comment:
There we are interested only in moves for out king so it is not necessary to generate all possible moves for our color